modules/currency-l1/src/test/scala/org/tessellation/currency/l1/http/DataApplicationRoutesSuite.scala

```diff
@@ -0,0 +1,118 @@
+package org.tessellation.currency.l1.http
+
+import cats.effect.std.{Queue, Random, Supervisor}
+import cats.effect.{IO, Resource}
+import cats.syntax.all._
+
+import org.tessellation.currency.dataApplication.{ConsensusInput, DataUpdate, L1NodeContext}
+import org.tessellation.currency.l1.MockDataApplicationL1Service
+import org.tessellation.currency.l1.MockDataApplicationState.{MockUpdate, mockUpdateGen}
+import org.tessellation.currency.l1.node.L1NodeContext
+import org.tessellation.currency.schema.currency.{CurrencyIncrementalSnapshot, CurrencySnapshotInfo}
+import org.tessellation.ext.cats.effect.ResourceIO
+import org.tessellation.json.JsonSerializer
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.node.shared.domain.queue.ViewableQueue
+import org.tessellation.node.shared.infrastructure.cluster.storage.L0ClusterStorage
+import org.tessellation.node.shared.infrastructure.snapshot.storage.LastSnapshotStorage
+import org.tessellation.schema.generators.{peerGen, signedOf}
+import org.tessellation.schema.peer.{L0Peer, Peer}
+import org.tessellation.schema.{GlobalIncrementalSnapshot, GlobalSnapshotInfo}
+import org.tessellation.security.signature.Signed
+import org.tessellation.security.{Hasher, JsonHash, SecurityProvider}
+import org.tessellation.shared.sharedKryoRegistrar
+
+import org.http4s.Method.GET
+import org.http4s.client.dsl.io._
+import org.http4s.implicits.http4sLiteralsSyntax
+import org.http4s.{HttpRoutes, Request, _}
+import org.scalacheck.Gen
+import suite.HttpSuite
+
+object DataApplicationRoutesSuite extends HttpSuite {
+
+  type Res = (SecurityProvider[IO], Hasher[IO], Supervisor[IO], Random[IO])
+
+  def sharedResource: Resource[IO, Res] = for {
+    implicit0(ks: KryoSerializer[IO]) <- KryoSerializer.forAsync[IO](sharedKryoRegistrar)
+    sp <- SecurityProvider.forAsync[IO]
+    implicit0(json2bin: JsonSerializer[IO]) <- JsonSerializer.forSync[IO].asResource
+    h: Hasher[IO] <- IO(Hasher.forSync[IO](_ => JsonHash)).asResource
+    sv <- Supervisor[IO]
+    r <- Random.scalaUtilRandom.asResource
+  } yield (sp, h, sv, r)
+
+  def construct(peer: Peer)(updateQueue: ViewableQueue[F, Signed[DataUpdate]]): IO[HttpRoutes[IO]] =
+    sharedResource.use { res =>
+      implicit val (sp, h, sv, r) = res
+      for {
+        lastGlobalSnapshotStorage <- LastSnapshotStorage.make[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo]
+        lastCurrencySnapshotStorage <- LastSnapshotStorage.make[F, CurrencyIncrementalSnapshot, CurrencySnapshotInfo]
+        implicit0(ctx: L1NodeContext[IO]) = L1NodeContext.make(lastGlobalSnapshotStorage, lastCurrencySnapshotStorage)
+        consensusQueue <- Queue.unbounded[F, Signed[ConsensusInput.PeerConsensusInput]]
+        l0ClusterStorage <- L0ClusterStorage.make(L0Peer.fromPeer(peer))
+        l1Service <- MockDataApplicationL1Service.make
```

Comment:
Do you really need all of these to be initialized properly? You may provide some stubs instead